### PR TITLE
Fix state=get on route53 module (Issue #423)

### PR DESCRIPTION
**Pull request by pjrm**
https://github.com/ansible-collections/community.aws/issues/423 This bug was introduced when refactoring from boto to boto3 library. This happens because the method "get_hosted_zone" only returns the DelegationSet when the DNS zone is external. Therefore this breaks when trying to get internal records.

The solution is to search for getting DNS records of type ''NS'' with the same name as the hosted zone.

Also, I added a test to cover this case.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
- route53

### CODE REVIEW

**Review by markuman — APPROVED** [review]

File: changelogs/fragments/406-route53-state-get.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- route53 - fix when using `state=get` on private DNS zones and add tests to cover this scenario (https://github.com/ansible-collections/community.aws/pull/424).
```
Comment (tremble) [inline changelogs/fragments/406-route53-state-get.yml:2]:
```suggestion
- route53 - fix when using ``state=get`` on private DNS zones and add tests to cover this scenario (https://github.com/ansible-collections/community.aws/pull/424).
```

**Review by tremble — APPROVED** [review]
Minor tweak to the changelog format, but otherwise looks good.

**Comment by tremble** [comment]
Tests are passing.  Thought I merged this already :)

**Comment by tremble** [comment]
@pjrm Thanks for your work on this issue.  Sorry it's taken a while to get this merged.

**Pull request merged**